platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -328,6 +336,15 @@ public class SwirldsPlatform implements Platform, Startable {
 
         registerAddressBookMetrics(metrics, initialAddressBook, selfId);
 
+        try {
+            recycleBin = new RecycleBin(platformContext.getConfiguration(), selfId);
+            if (softwareUpgrade) {
+                recycleBin.clear();
```

Comment:
files from a previous version are no longer relevant?

Reply:
There is no well defined length of time that a file will be relevant, other than older files are less likely to be relevant than newer files. This is mostly just a heuristic shortcut. The upgrade boundaries are a very convenient hook that clears out this directory at least once per month.

Eventually, we will implement a better algorithm, but that is going to require a lot more work than this simple algorithm did. https://github.com/hashgraph/hedera-services/issues/6687